tests/unit_tests/test_wave_06.py

```diff
@@ -110,7 +112,13 @@ def test_swap_best_by_category():
         their_priority="Decor"
     )
 
-    raise Exception("Complete this test according to comments below.")
+    assert result
+    assert len(tai.inventory) == 3
+    assert len(jesse.inventory) == 3
+    assert tai.get_by_id(5678) != None
+    assert jesse.get_by_id(1234) != None
```

Comment:
I think this is a really cool way to test, and I encourage you to keep doing this whenever you find it appropriate.

As a side note, in your approach, this test will _only_ work if we can trust that `get_by_id` is working accurately (aka this test has a _dependency_ on `get_by_id` working). I'm only pointing this out so, if there's ever a time where you/another developer ever finds this test to be flaky/broken/unreliable for any reason, you/another developer can quickly think, "I should also make sure that `get_by_id` is also working."

I'm definitely describing a pretty specific situation that I've come across in industry-- if you're curious about this more, let me know and we can talk about it!